kim/kim-ldap/src/main/java/edu/arizona/kim/service/impl/UaLdapIdentityServiceImpl.java

```diff
@@ -0,0 +1,777 @@
+import org.kuali.rice.kim.api.identity.entity.EntityDefault;
+import org.kuali.rice.kim.api.identity.entity.EntityDefaultQueryResults;
+import org.kuali.rice.kim.api.identity.entity.EntityQueryResults;
+import org.kuali.rice.kim.api.identity.external.EntityExternalIdentifier;
+import org.kuali.rice.kim.api.identity.external.EntityExternalIdentifierType;
+import org.kuali.rice.kim.api.identity.name.EntityName;
+import org.kuali.rice.kim.api.identity.personal.EntityBioDemographics;
+import org.kuali.rice.kim.api.identity.personal.EntityEthnicity;
+import org.kuali.rice.kim.api.identity.phone.EntityPhone;
+import org.kuali.rice.kim.api.identity.principal.EntityNamePrincipalName;
+import org.kuali.rice.kim.api.identity.principal.Principal;
+import org.kuali.rice.kim.api.identity.principal.PrincipalQueryResults;
+import org.kuali.rice.kim.api.identity.privacy.EntityPrivacyPreferences;
+import org.kuali.rice.kim.api.identity.residency.EntityResidency;
+import org.kuali.rice.kim.api.identity.type.EntityTypeContactInfo;
+import org.kuali.rice.kim.api.identity.visa.EntityVisa;
+import org.kuali.rice.kim.dao.LdapPrincipalDao;
+import org.kuali.rice.kim.impl.KIMPropertyConstants;
+import org.kuali.rice.kim.impl.identity.EntityTypeBo;
+import org.kuali.rice.kim.impl.identity.address.EntityAddressTypeBo;
+import org.kuali.rice.kim.impl.identity.affiliation.EntityAffiliationTypeBo;
+import org.kuali.rice.kim.impl.identity.citizenship.EntityCitizenshipStatusBo;
+import org.kuali.rice.kim.impl.identity.email.EntityEmailTypeBo;
+import org.kuali.rice.kim.impl.identity.employment.EntityEmploymentStatusBo;
+import org.kuali.rice.kim.impl.identity.employment.EntityEmploymentTypeBo;
+import org.kuali.rice.kim.impl.identity.external.EntityExternalIdentifierTypeBo;
+import org.kuali.rice.kim.impl.identity.name.EntityNameTypeBo;
+import org.kuali.rice.kim.impl.identity.phone.EntityPhoneTypeBo;
+import org.kuali.rice.kim.service.LdapIdentityService;
+import org.kuali.rice.krad.service.BusinessObjectService;
+
+import java.util.*;
+
+/**
+ * This class is a simplified implementation of the IdentityService and
+ * LdapIdentityService interfaces. The vanilla IdentityService impls are more
+ * complex in their inheritance and composition. Alternatively, this class
+ * provides a clear and concise hierarchy which then serves to reduce bugs.
+ * This implementation also cuts out inheritance to classes that still contain
+ * termination points to KIM tables -- i.e., this class is gaurunteed to
+ * *always* fetch its entity records directly from the EDS implementation of LDAP.
+ *
+ * Any methods that do not line up with this LDAP paradigm, have all been implemented
+ * to throw an UnsupportedOperationException. This works since all of the screens
+ * in the UI that can mutate Person data, are all turned off here at the UA. In such
+ * a way, we ensure non-LDAP operations are never executed.
+ */
+public class UaLdapIdentityServiceImpl implements LdapIdentityService {
```

Comment:
UA Ldap Identity service is not using inheritance to get implemented methods from LDAPIdentityDelegateServiceImpl, but instead, it provides its own implementation of the LdapIdentiyService interface.